src/utils.ts

```diff
@@ -1,5 +1,10 @@
+//# #if HAVE_VSCODE
 import type { LSPObject } from "vscode-languageclient";
 import variables from "vscode-variables";
+//# #elif HAVE_COC_NVIM
```

Comment:
I want to have these macros replaced with dynamic code which can be vetted with linters. is that possible?

Reply:
after this PR, We can add a CI/CD to run `scripts/patch.sh ../src/utils.ts` then use linter to check it.